File: Gestion_Evenement2/src/main/java/edu/esprit/entities/Question.java
package edu.esprit.entities;

public class Question {
    private int idQ;
    private int idQz;
    private String questionText;
    private String choix_1;
    private String choix_2;
    private String choix_3;
    private String choix_4;
    private String choix_correcte;

    public Question(){idQ = 0;}
    public Question(int idQ, int idQz, String questionText, String choix_1, String choix_2, String choix_3, String choix_4, String choix_correcte) {
        this.idQ = idQ;
        this.idQz = idQz;
        this.questionText = questionText;
        this.choix_1 = choix_1;
        this.choix_2 = choix_2;
        this.choix_3 = choix_3;
        this.choix_4 = choix_4;
        this.choix_correcte = choix_correcte;
    }

    public int getIdQ() {
        return idQ;
    }

    public void setIdQ(int idQ) {
        this.idQ = idQ;
    }

    public int getIdQz() {
        return idQz;
    }

    public void setIdQz(int idQz) {
        this.idQz = idQz;
    }

    public String getQuestionText() {
        return questionText;
    }

    public void setQuestionText(String questionText) {
        this.questionText = questionText;
    }

    public String getChoix_1() {
        return choix_1;
    }

    public void setChoix_1(String choix_1) {
        this.choix_1 = choix_1;
    }

    public String getChoix_2() {
        return choix_2;
    }

    public void setChoix_2(String choix_2) {
        this.choix_2 = choix_2;
    }

    public String getChoix_3() {
        return choix_3;
    }

    public void setChoix_3(String choix_3) {
        this.choix_3 = choix_3;
    }

    public String getChoix_4() {
        return choix_4;
    }

    public void setChoix_4(String choix_4) {
        this.choix_4 = choix_4;
    }

    public String getChoix_correcte() {
        return choix_correcte;
    }

    public void setChoix_correcte(String choix_correcte) {
        this.choix_correcte = choix_correcte;
    }
}

File: projettest-main2/src/Entity/Question.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;


use App\Repository\QuestionRepository;

#[ORM\Entity(repositoryClass: QuestionRepository::class)]
class Question
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idq = null;

    #[ORM\Column(length: 255)]
    private $questiontext;

    #[ORM\Column(length: 255)]
    private $choix1;

    #[ORM\Column(length: 255)]
    private $choix2;

    #[ORM\Column(length: 255)]
    private $choix3;

    #[ORM\Column(length: 255)]
    private $choix4;

    #[ORM\Column(length: 255)]
    private $choixCorrecte;


    #[ORM\ManyToOne(targetEntity:Quiz::class , inversedBy:' Question')]
    private ?Quiz $idqz;

    public function getIdq(): ?int
    {
        return $this->idq;
    }

    public function getQuestiontext(): ?string
    {
        return $this->questiontext;
    }

    public function setQuestiontext(?string $questiontext): static
    {
        $this->questiontext = $questiontext;

        return $this;
    }

    public function getChoix1(): ?string
    {
        return $this->choix1;
    }

    public function setChoix1(string $choix1): static
    {
        $this->choix1 = $choix1;

        return $this;
    }

    public function getChoix2(): ?string
    {
        return $this->choix2;
    }

    public function setChoix2(string $choix2): static
    {
        $this->choix2 = $choix2;

        return $this;
    }

    public function getChoix3(): ?string
    {
        return $this->choix3;
    }

    public function setChoix3(string $choix3): static
    {
        $this->choix3 = $choix3;

        return $this;
    }

    public function getChoix4(): ?string
    {
        return $this->choix4;
    }

    public function setChoix4(string $choix4): static
    {
        $this->choix4 = $choix4;

        return $this;
    }

    public function getChoixCorrecte(): ?string
    {
        return $this->choixCorrecte;
    }

    public function setChoixCorrecte(string $choixCorrecte): static
    {
        $this->choixCorrecte = $choixCorrecte;

        return $this;
    }

   /* public function getIdqz(): ?Quiz
    {
        return $this->idqz;
    }*/

    public function setIdqz(?Quiz $idqz): static
    {
        $this->idqz = $idqz;

        return $this;
    }


}